File: FE-main/src/types/Project.ts
export interface ICreateProject {
    name: string;
    description: string;
    startDate: Date;
    endDate: Date;
    scale: boolean;
}

export interface IProject {
    pk: number;
    fields: {
        date_begin: string;
        date_end: string;
        description: string;
        name: string;
        owner_id: number;
        status: string;
        scale_risk: boolean;
    };
    model?: string;
}

export interface IProjectRole {
    pk: number;
    model?: string;
    fields: {
        project: number;
        role: string;
        user: number;
    };
}

File: app/models/project.py
from django.db import models
from .user import User
from .choices import UserProjectRoles, Status
import datetime


# Class Project
class Project(models.Model):
    owner_id = models.ForeignKey(User, null=True, on_delete=models.SET_NULL)
    name = models.CharField(max_length=256, null=True)
    description = models.CharField(max_length=512, blank=True, null=True)
    status = models.CharField(max_length=64, choices=Status.choices, default=Status.CONCEPT)
    scale_risk = models.BooleanField(default=True)
    date_begin = models.DateField(default=datetime.date.today)
    date_end = models.DateField(null=True)

    def __str__(self):
        return self.name


# Class UserProject
class UserProject(models.Model):
    user = models.ForeignKey(User, on_delete=models.CASCADE)
    project = models.ForeignKey(Project, on_delete=models.CASCADE)
    role = models.CharField(max_length=64, choices=UserProjectRoles.choices)

    def __str__(self):
        return self.user.name + ' (' + self.role + ') - ' + self.project.name